Improvement/arsn 423 bump dependencies

benzekrimaha opened the pull request:
Issue : ARSN-423

bert-e commented:
# Hello benzekrimaha,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/arsenal/doc/user).


<details>
  <summary><b>Available options</b></summary>

  name   | description  | privileged   | authored
  ------ | ------------ | ------------ |----------
  `/after_pull_request` | Wait for the given pull request id to be merged before continuing with the current one. |  | 
  `/bypass_author_approval` | Bypass the pull request author's approval |  :star:  | 
  `/bypass_build_status` | Bypass the build and test status |  :star:  | 
  `/bypass_commit_size` | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
  `/bypass_incompatible_branch` | Bypass the check on the source branch prefix |  :star:  | 
  `/bypass_jira_check` | Bypass the Jira issue check |  :star:  | 
  `/bypass_peer_approval` | Bypass the pull request peers' approval |  :star:  | 
  `/bypass_leader_approval` | Bypass the pull request leaders' approval |  :star:  | 
  `/approve` | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
  `/create_pull_requests` | Allow the creation of integration pull requests. |  | 
  `/create_integration_branches` | Allow the creation of integration branches. |  | 
  `/no_octopus` | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
  `/unanimity` | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
  `/wait` | Instruct Bert-E not to run until further notice. |  | 
  

</details>



<details>
  <summary><b>Available commands</b></summary>

  name   | description  | privileged
  ------ | ------------ | ------------
  `/help` | Print Bert-E's manual in the pull request. | 
  `/status` | Print Bert-E's current status in the pull request ```TBA``` | 
  `/clear` | Remove all comments from Bert-E from the history ```TBA``` | 
  `/retry` | Re-start a fresh build ```TBA``` | 
  `/build` | Re-start a fresh build ```TBA``` | 
  `/force_reset` | Delete integration branches & pull requests, and restart merge process from the beginning. | 
  `/reset` | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
  
</details>




*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ARSN-423 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `8.2.0`


Please check the `Fix Version/s` of ARSN-423, or the target
branch of this pull request.

bert-e commented:
# Conflict




There is a conflict between your branch `improvement/ARSN-423-bump-dependencies` and the
destination branch `development/8.2`.

Please resolve the conflict on **the feature branch** (`improvement/ARSN-423-bump-dependencies`).

```sh
git fetch && \
git checkout origin/improvement/ARSN-423-bump-dependencies && \
git merge origin/development/8.2
```

Resolve merge conflicts and commit

```sh
git push origin HEAD:improvement/ARSN-423-bump-dependencies
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

benzekrimaha committed:
Bump dependencies

Issue: ARSN-423

benzekrimaha committed:
Adapt config post bump eslint to v9

Issue: ARSN-423

benzekrimaha committed:
Update code to match scality guidelines and dependencies bumps

In this commit the code has been updated to match the new
uuid require based on the documentation here : https://www.npmjs.com/package/uuid
timestamp format has been updated as well from new Timestamp(1, 1651144629)
to new Timestamp({ t: 1651144629 , i: 1 }), the remaining changes are
lint fixups ( please note that the indentation has been fixed as well automatically
using lint --fix)
Issue: ARSN-423

benzekrimaha committed:
Tests fixups after CI failures due to bumps

In this commit we fix the tests that were failing after the dependencies bumps.
Issue: ARSN-423

benzekrimaha committed:
Bump node version in gha

Issue: ARSN-423

williamlardier commented on lib/s3middleware/processMpuParts.ts:
```diff
@@ -86,7 +86,7 @@ export function generateMpuPartStorageInfo(filteredPartList: any[]) {
             };
             dataLocations.push(pieceRetrievalInfo);
-            // eslint-disable-next-line no-param-reassign
+             
```
```suggestion
```

williamlardier commented on lib/policyEvaluator/utils/wildcards.ts:
```diff
@@ -30,7 +30,7 @@ export const handleWildcards = (string: string) => {
     // Then replace the AWS special characters with regExp equivalents
     const regExStr = string.replace(/[\\^$*+?.()|[\]{}]/g, '\\$&').replace(
-        // eslint-disable-next-line max-len
+         
```
```suggestion
```

williamlardier commented on lib/policyEvaluator/utils/conditions.ts:
```diff
@@ -279,9 +277,9 @@ export function convertConditionOperator(operator: string): boolean {
             } else {
                 return policyValRegex(key);
             }
+            return true;
```
I think we should either return undefined here, because if we reach this stage, it means the "key" was an array and a prefix that is unknown...

williamlardier commented on tests/unit/algos/list/delimiterMaster.spec.ts:
```diff
@@ -1,4 +1,4 @@
-'use strict'; // eslint-disable-line strict
+'use strict';  
```
```suggestion
'use strict';
```

williamlardier commented on tests/functional/clustering/ClusterRPC.spec.js:
```diff
@@ -93,7 +93,7 @@ function runTest(testUrl, cb) {
 }
 
-describe('ClusterRPC', () => {
+describe.only('ClusterRPC', () => {
```
```suggestion
describe('ClusterRPC', () => {
```

williamlardier commented on package.json:
```diff
@@ -17,76 +17,78 @@
     "lint_md": "mdlint $(git ls-files '*.md')",
     "lint_yml": "yamllint $(git ls-files '*.yml')",
-    "test": "jest tests/unit",
+    "test": "export NODE_OPTIONS=\"--tls-max-v1.2\" && jest tests/unit --detectOpenHandles",
```
We should set this option from github actions files, and eventually mention it in a readme or doc, we shouldn't need to change this command

francoisferrand replied:
The flag was added to package.json on purpose, so devepopers can be run locally as well.

As for the ticket we definitely need one, and it needs to done ASAP... though I'd prefer we merge this PR (no problem as long as it's not released, it will land on a new dev branch) and fix TLS 1.3 soon after, to avoid more churning on the node migration (rebase, conflicts...)

williamlardier commented on lib/versioning/Version.ts:
```diff
@@ -111,9 +110,9 @@ export class Version {
-            (needComma ? ',' : '') +
-            `"${key}":"${value}"}`
+            `${stringifiedObject.slice(0, stringifiedObject.length - 1)}${ 
+                needComma ? ',' : '' 
```
```suggestion
                needComma ? ',' : ''
```

williamlardier commented on lib/s3routes/routesUtils.ts:
```diff
@@ -495,6 +493,7 @@ function _computeContentLengthFromLocation(
                 return sum + parseInt(location.size, 10);
             }
         }
+        return sum;
```
As sum is undefined here,

```suggestion
        return;
```

williamlardier commented on lib/s3middleware/userMetadata.ts:
```diff
@@ -11,7 +11,7 @@ export function getMetaHeaders(headers: http.IncomingHttpHeaders) {
     const filtered = rawHeaders.filter(([k]) => k.startsWith('x-amz-meta-'));
     const totalLength = filtered.reduce((length, [k, v]) => {
-        if (!v) return length;
+        if (!v) {return length;}
```
```suggestion
        if (!v) {
            return length;
        }
```

williamlardier commented on lib/models/ObjectMDArchive.ts:
```diff
@@ -51,7 +51,7 @@ export default class ObjectMDArchive {
     }) {
         try {
-            // eslint-disable-next-line no-new
+             
```
```suggestion
```

williamlardier commented on lib/models/ObjectMDAmzRestore.ts:
```diff
@@ -32,10 +32,10 @@ export default class ObjectMDAmzRestore {
     static isValid(data: { 'ongoing-request': boolean; 'expiry-date': Date | string }) {
         try {
-            // eslint-disable-next-line no-new
+             
```
```suggestion
```

williamlardier commented on lib/models/LifecycleConfiguration.ts:
```diff
@@ -667,15 +667,15 @@ export default class LifecycleConfiguration {
+            ':([0-5][0-9])' + // Second
+            '(\\.[0-9]+)?' + // Fractional second
+            '(Z|[+-][01][0-9]:[0-5][0-9])?$', // Timezone
+            'g'
```
```suggestion
            'g',
```

williamlardier commented on lib/models/LifecycleConfiguration.ts:
```diff
@@ -262,7 +262,7 @@ export default class LifecycleConfiguration {
             } else {
                 const propName = prop.propName;
-                // eslint-disable-next-line no-param-reassign
+                 
```
```suggestion
```

williamlardier commented on lib/algos/list/delimiter.ts:
```diff
@@ -1,4 +1,4 @@
-'use strict'; // eslint-disable-line strict
+'use strict';  
```
```suggestion
'use strict';
```

benzekrimaha committed:
Post reviews fixups

KillianG commented on .github/workflows/tests.yaml:
```diff
@@ -28,11 +28,10 @@ jobs:
     - uses: actions/setup-node@v4
       with:
-        node-version: '16'
+        node-version: '20'
```
Can we test with node 22 ? (even though we enforce node 20 in the package.json) should be great to test with latest version

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

bert-e commented:
# Conflict




There is a conflict between your branch `improvement/ARSN-423-bump-dependencies` and the
destination branch `development/8.2`.

Please resolve the conflict on **the feature branch** (`improvement/ARSN-423-bump-dependencies`).

```sh
git fetch && \
git checkout origin/improvement/ARSN-423-bump-dependencies && \
git merge origin/development/8.2
```

Resolve merge conflicts and commit

```sh
git push origin HEAD:improvement/ARSN-423-bump-dependencies
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

bert-e commented:
# Conflict




There is a conflict between your branch `improvement/ARSN-423-bump-dependencies` and the
destination branch `development/8.2`.

Please resolve the conflict on **the feature branch** (`improvement/ARSN-423-bump-dependencies`).

```sh
git fetch && \
git checkout origin/improvement/ARSN-423-bump-dependencies && \
git merge origin/development/8.2
```

Resolve merge conflicts and commit

```sh
git push origin HEAD:improvement/ARSN-423-bump-dependencies
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

benzekrimaha commented:
/approve

benzekrimaha commented:
/wait

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers











*The following options are set:* **approve**

KillianG commented on package.json:
```diff
@@ -17,72 +17,75 @@
+    "joi": "^17.13.3",
     "level": "~5.0.1",
     "level-sublevel": "~6.6.5",
     "mongodb": "^5.2.0",
```
shouldn't we bump mongodb to 6.1 ?

benzekrimaha committed:
bump mongodb and adapt code

williamlardier commented on lib/storage/metadata/mongoclient/LogConsumer.js:
```diff
@@ -35,27 +36,29 @@ class LogConsumer {
-                    { error: err });
-                return done(err);
-            }
+            useUnifiedTopology: true,
```
This shouldn't be needed, I guess this one is deprecated and at least enabled by default, can we check?

williamlardier commented on lib/s3routes/routesUtils.ts:
```diff
@@ -147,8 +147,8 @@ export const XMLResponseBackend = {
             '<Error>',
-            `<Code>${errCode.message}</Code>`,
-            `<Message>${errCode.description}</Message>`,
+            `<Code>${error.message}</Code>`,
+            `<Message>${error.description}</Message>`,
         );
```
I believe this was changed already in 8.1, you can rebase your branch

benzekrimaha replied:
The branch has been rebased

williamlardier commented on lib/network/rpc/sio-stream.ts:
```diff
@@ -415,7 +415,7 @@ class SIOStreamSocket {
                 // user objects are simple flat objects and we want to
                 // copy all their properties
-                // eslint-disable-next-line
+                 
```
```suggestion
```

williamlardier commented on lib/network/rpc/sio-stream.ts:
```diff
@@ -341,7 +341,7 @@ class SIOStreamSocket {
                 // user objects are simple flat objects and we want to
                 // copy all their properties
-                // eslint-disable-next-line
+                 
```
```suggestion
```

williamlardier commented on tests/unit/policyEvaluator/RequestContext.spec.js:
```diff
@@ -126,4 +125,46 @@ describe('RequestContext', () => {
+        const ssoRC = new RequestContext(...ssoParams);
+        assert.strictEqual(ssoRC.getResource(), 'arn:scality:sso:::general-resource/specific-resource');
+    });
+});
```
```suggestion
});

```

francoisferrand commented on .github/workflows/tests.yaml:
```diff
@@ -44,6 +43,8 @@ jobs:
     - name: test and coverage
       run: yarn --silent coverage
+      env:
+        NODE_OPTIONS: "--tls-max-v1.2"
```
as we need the options to pass the test:
* should it not be set instead in package.json instead, so the test can be run locally?
* do we need to do this kind of thing in the actual services (backbeat, cloudserver) as well, or is it really only affecting tests?

benzekrimaha replied:
For the moment it's only affecting tests , indeed to be run locally we can put this in the test command

francoisferrand commented on README.md:
```diff
@@ -55,6 +55,11 @@ console.log(errors.AccessDenied);
 
+## Build Status
+
+![Public Build Status][badgepub]
+![Private Build Status][badgepriv]
+
```
+1 for adding badges, but these badges do not seem relevant:

```
[badgepub]: https://circleci.com/gh/scality/Arsenal.svg?style=svg
[badgepriv]: http://ci.ironmann.io/gh/scality/Arsenal.svg?style=svg&circle-token=c3d2570682cba6763a97ea0bc87521941413d75c
```

Either do not add the badges, or actually fix them, based on gha and codecov.
Not sure what the "private build status" is.

benzekrimaha replied:
Done here : https://github.com/scality/Arsenal/pull/2266/commits/33e77f04b6684f30618f9f382ba82461bbef11a5

francoisferrand commented on tsconfig.json:
```diff
@@ -13,7 +13,8 @@
         "noEmitOnError": false,
-        "sourceMap": true
+        "sourceMap": true,
+        "skipLibCheck": true,
```
why do we disable lib check?
i understand it may be useful when doing the bump (faster ts build), but generalyl we should keep it (and "resolve" dependency if multiple copies of the same lib are present) ?

francoisferrand commented on lib/network/kmip/Client.ts:
```diff
@@ -364,7 +363,7 @@ export default class Client {
         ], (err, response) => {
             if (err) {
-                const error = arsenalErrorKMIP(err);
+                const error = arsenalErrorKMIP(JSON.stringify(err));
```
do we really want to stringify: should not just be `err.message` ?
and/or is there a risk of leaking details to the (s3 api) caller?

francoisferrand commented on README.md:
```diff
@@ -147,4 +155,4 @@ to the process (which results in an immediate termination, and this signal
 can't be caught).
 
 [badgepub]: https://circleci.com/gh/scality/Arsenal.svg?style=svg
```
this badge is not relevant anymore: we are not using circleci anymore, but only github actions.

c.f. https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/monitoring-workflows/adding-a-workflow-status-badge

francoisferrand replied:
not adressed yet (for the `badgepub` badge) : still referebnces `https://circleci.com`

benzekrimaha replied:
Done here : https://github.com/scality/Arsenal/pull/2266/commits/33e77f04b6684f30618f9f382ba82461bbef11a5

francoisferrand commented on lib/network/kmip/transport/TransportTemplate.ts:
```diff
@@ -90,9 +90,14 @@ export default class TransportTemplate {
         try {
+            const options = {
+                ...this.options.tls,
+                host: '127.0.0.1',
```
this code is used in production, we should not set the host here (esp. it overrides `this.optins.tls`)
--> should be set in the tests i guess?

francoisferrand commented on lib/network/kmip/transport/TransportTemplate.ts:
```diff
@@ -90,9 +90,14 @@ export default class TransportTemplate {
+            const options = {
+                ...this.options.tls,
+                host: '127.0.0.1',
+                family: 4,         
```
is there a constant somewhere, instead of using `4` ?
otherwise, can we at least use the string representation instead ('IPv4') ?

benzekrimaha committed:
changing ubuntu image to fix dependency issue

benzekrimaha committed:
package.json fixups

francoisferrand commented on lib/policyEvaluator/requestUtils.ts:
```diff
@@ -19,7 +19,9 @@ export interface S3Config {
     const remoteAddress = request.socket.remoteAddress;
-    const clientIp = remoteAddress?.toString() ?? '';
+    // TODO What to do if clientIp === undefined ?
+    const clientIp = (requestConfig ? remoteAddress : request.headers['x-forwarded-for'] ||
```
bad merge?
`x-forwarded-for` should not be used unless a requestConfig is provided, to allow validating that the header comes from a proxy that is indeed "allowed" to provide us this header (otherwise anybody could impersonate any IP address using this header)

benzekrimaha replied:
it has been fixed , indeed it was a bad merge i was fixing

welansari commented on lib/auth/Vault.ts:
```diff
@@ -36,7 +36,7 @@ function vaultSignatureCb(
         return callback(err);
     }
 
-    const { email, ...userInfoWithoutEmail } = authInfo.message.body.userInfo;
+    const { ...userInfoWithoutEmail } = authInfo.message.body.userInfo;
 
     log.debug('received info from Vault', {
         ...authInfo,
```
The goal of this was to exclude the email from the info we set in `userInfo` below, with the new change it's included again.

benzekrimaha replied:
No the goal was not to exclude it , the value was declared but never used thus causing a lint error

welansari replied:
I'm pretty sure the email was removed from the log for security reasons, you check this commit https://github.com/scality/Arsenal/commit/6374fc831d9416c285a1d96ca18f614533ee8e5d (the name of the variable also suggests it)

williamlardier replied:
We can try another approach if we want to avoid the linting issue but yes, we should not print PII in the logs, and the auth info has some (the email)

francoisferrand replied:
this should do it ?

```suggestion
    const { email: _, ...userInfoWithoutEmail } = authInfo.message.body.userInfo;

    log.debug('received info from Vault', {
        ...authInfo,
```

francoisferrand commented on README.md:
````diff
@@ -55,6 +53,14 @@ console.log(errors.AccessDenied);
 
 ```
 
+## Build Status
+
+![badgepub]
+
+## Codecov Status
+
+![codecov]
````
i think it's the usual practice to keep the badges "displayed" on top of the README file, so they are visible when we open the github repository's page

francoisferrand commented on README.md:
```diff
@@ -146,5 +152,5 @@ before the application has finished it's cleanup, a 'SIGKILL' signal is send
-[badgepub]: https://circleci.com/gh/scality/Arsenal.svg?style=svg
-[badgepriv]: http://ci.ironmann.io/gh/scality/Arsenal.svg?style=svg&circle-token=c3d2570682cba6763a97ea0bc87521941413d75c
+[badgepub]: https://github.com/scality/Arsenal/actions/workflows/tests.yaml/badge.svg
+[codecov]: https://codecov.io/gh/scality/Arsenal/branch/development/8.1/graph/badge.svg?token=X0esXhJSwb
```
is this token "safe" to share here: where does it come from/what does it give access to?

Just tried without the token, and it seems to work jsut fine: even in a new "private browsing" window, so i guess we can just remove it...

```suggestion
[codecov]: https://codecov.io/gh/scality/Arsenal/branch/development/8.1/graph/badge.svg
```

francoisferrand commented on lib/storage/metadata/mongoclient/MongoClientInterface.ts:
```diff
@@ -1682,7 +1686,7 @@ class MongoClientInterface {
                 // atomic condition on the PHD flag and the mst
                 // version:
-                // eslint-disable-next-line
+                 
```
uneeded empty line (between a comment and the code it refers to) + space at end of line

```suggestion
```

francoisferrand commented on lib/storage/metadata/mongoclient/MongoClientInterface.ts:
```diff
@@ -2585,15 +2588,15 @@ class MongoClientInterface {
-                    return cb(err);
-                }
-                return cb(errors.InternalError);
+            log.error('could not get list of collections', {
+                method: 'getBucketInfos',
+                error: err,
             });
+            if (err && err instanceof ArsenalError) {
+                return cb(err);
+            }
+            return cb(errors.InternalError);
+        });
     }
```
bad indent
```suggestion
                log.error('could not get list of collections', {
                    method: 'getBucketInfos',
                    error: err,
                });
                if (err && err instanceof ArsenalError) {
                    return cb(err);
                }
                return cb(errors.InternalError);
            });
    }
```

or wrap `).catch(err => {` to better show that there are 2 "nested" promises (catch relates to `toArray().then`, not to `async.eachLimit`)
```suggestion
            })
        ).catch(err => {
            log.error('could not get list of collections', {
                method: 'getBucketInfos',
                error: err,
            });
            if (err && err instanceof ArsenalError) {
                return cb(err);
            }
            return cb(errors.InternalError);
        });
```

francoisferrand commented on lib/storage/metadata/mongoclient/MongoClientInterface.ts:
```diff
@@ -2630,22 +2633,20 @@ class MongoClientInterface {
 
     consolidateData(store, dataManaged) {
-        /* eslint-disable */
+         
```
extra empty line + space at end of line

francoisferrand commented on lib/storage/metadata/mongoclient/MongoClientInterface.ts:
```diff
@@ -2977,20 +2977,20 @@ class MongoClientInterface {
             }).then(() => {
-                const bucketStatus = bucketInfo.getVersioningConfiguration();
-                const isVer = (bucketStatus &&
+            const bucketStatus = bucketInfo.getVersioningConfiguration();
```
same here, weird indent : should wrap `then` ?

francoisferrand commented on lib/storage/metadata/mongoclient/MongoClientInterface.ts:
```diff
@@ -574,7 +578,7 @@ class MongoClientInterface {
         // Quota must be stored as a Long to account for values larger than
         // Number.MAX_SAFE_INTEGER.
-        // eslint-disable-next-line new-cap
+         
```
nit: extra empty line (and spaces at EOL)

```suggestion
```

francoisferrand commented on README.md:
```diff
@@ -1,6 +1,7 @@
 # Arsenal
 
-[![codecov](https://codecov.io/gh/scality/Arsenal/branch/development/8.1/graph/badge.svg?token=X0esXhJSwb)](https://codecov.io/gh/scality/Arsenal)
+[![Public Build Status](https://github.com/scality/Arsenal/actions/workflows/tests.yaml/badge.svg)](https://github.com/scality/Arsenal/actions/workflows/tests.yaml)
+[![codecov](https://codecov.io/gh/scality/Arsenal/branch/development/8.1/graph/badge.svg)](https://codecov.io/gh/scality/Arsenal)
```
nit: can we reuse the badges defined at the end?

```suggestion
[![badgepub]](https://github.com/scality/Arsenal/actions/workflows/tests.yaml)
[![codecov]](https://codecov.io/gh/scality/Arsenal)
```

Seragonia committed:
Bump to 8.2.0

Issue: ARSN-423

francoisferrand committed:
Rename eslint-config-scality to @scality org

This avoids the malware advisory published by Github [1].

[1] https://github.com/advisories/GHSA-52g9-9f8x-g5m5

Issue: ARSN-423

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ARSN-423 contains:


* `7.10.66`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `8.2.0`


Please check the `Fix Version/s` of ARSN-423, or the target
branch of this pull request.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/8.2`



The following branches have **NOT** changed:

* `development/6.4`
* `development/7.10`
* `development/7.4`
* `development/7.70`
* `development/8.1`



Please check the status of the associated issue ARSN-423.

Goodbye benzekrimaha.




*The following options are set:* **approve**
